gym-unity/gym_unity/envs/__init__.py

```diff
@@ -180,25 +185,23 @@ def step(self, action: List[Any]) -> GymStepResult:
             return self._single_step(decision_step)
 
     def _single_step(self, info: Union[DecisionSteps, TerminalSteps]) -> GymStepResult:
-        if self.use_visual:
+        if self._allow_multiple_obs:
             visual_obs = self._get_vis_obs_list(info)
-
-            if self._allow_multiple_visual_obs:
-                visual_obs_list = []
-                for obs in visual_obs:
-                    visual_obs_list.append(self._preprocess_single(obs[0]))
-                self.visual_obs = visual_obs_list
-            else:
-                self.visual_obs = self._preprocess_single(visual_obs[0][0])
-
+            visual_obs_list = []
+            for obs in visual_obs:
+                visual_obs_list.append(self._preprocess_single(obs[0]))
+            self.visual_obs = visual_obs_list
             default_observation = self.visual_obs
-        elif self._get_vec_obs_size() > 0:
-            default_observation = self._get_vector_obs(info)[0, :]
+            if self._get_vec_obs_size() >= 1:
+                default_observation.append(self._get_vector_obs(info)[0, :])
```

Comment:
This will modify `self.visual_obs` with a non-visual obs - is that a problem?

Reply:
I think it matters since `self.visual_obs` is used in `env.render`. So I made it so it is always the first observation when many.